### PR TITLE
have letsencrypt.VERSION, show it in letsencrypt --help, use it in setup.py

### PR DESCRIPTION
**Pull request by ThomasWaldmann**
note: we had some discussion about potential problems importing VERSION from main package.

SO link: http://stackoverflow.com/questions/2058802/how-can-i-get-the-version-defined-in-setup-py-setuptools-in-my-package

See also my comment in __init__.py - maybe we can add that "version detection from git tags" magic later.

### CODE REVIEW

**Comment by ThomasWaldmann** [comment]
maybe nice also: -V, --version option that just gives the version.

**Comment by kuba** [comment]
My basic tests worked, though I still have a lot of reservation for this approach, c.f. zooko's comment over at SO regarding `py2exe` et al. It would be sad if we broke some (popular) ecosystems.

NB, another [SO answer](http://stackoverflow.com/questions/17583443/what-is-the-correct-way-to-share-package-version-with-setup-py-and-the-package) also suggests `pkg_resources`

What do others use? For example,

- [pypa](https://bitbucket.org/pypa) themselves (maintainers of `setuptools`)  use [`read`+`exec`](https://bitbucket.org/pypa/setuptools/src/9db269094f1dd2b79591bdfd95f1e759b647acda/setup.py?at=default#cl-22) or [`read + re`](https://bitbucket.org/pypa/wheel/src/bdf053a70200c5857c250c2044a2d91da23db4a9/setup.py?at=default#cl-11)
- https://github.com/Pylons projects use `pkg_resources`
- django uses [`__import__('django').get_version()`](https://github.com/django/django/blob/18f3e79b13947de0bda7c985916d5a04e28936dc/setup.py#L32), while requests uses [`import requests`](https://github.com/kennethreitz/requests/blob/359659cf4b9dbeeef1ed832501dc1f99b0f0beac/setup.py#L6)

Most of the projects I've seen do the same as pypa folks. I think I trust this solution the most. What do you think?

**Comment by jdkasten** [comment]
I didn't do any research myself, but I would be inclined to follow pypa approach.

**Comment by ThomasWaldmann** [comment]
updated.

File: setup.py [hunk 1]
```diff
@@ -1,7 +1,26 @@
 #!/usr/bin/env python
+import os
+import re
+import codecs
```
Comment (kuba) [inline setup.py]: `codes` should go a the top

File: setup.py [hunk 2]
```diff
@@ -1,7 +1,26 @@
...
+
 from setuptools import setup
 
 
+def read_file(filename, encoding='utf8'):
+    """read unicode from given file"""
```
Comment (kuba) [inline setup.py]: It doesn't matter that much here, but `__doc__` should start with capital letter, as in sentence. It looks nicer in docs. :)

**Comment by kuba** [comment]
LGTM, just pls fix above comments

**Comment by ThomasWaldmann** [comment]
fixed.

**Comment by jdkasten** [comment]
Thanks @ThomasWaldmann!

**Pull request merged**